corehq/apps/hqcase/case_helper.py

```diff
@@ -163,6 +167,112 @@ def close(
             is_creation=False,
         )
 
+    @classmethod
+    def copy_cases(
+        cls,
+        domain,
+        case_ids,
+        to_owner,
+        censor_data=None,
+        count_only=False
+    ):
+        """
+        Copies the cases governed by 'case_ids' to the respective 'to_owner'
+        and replacing the relevant case properties with the specified 'replace_props'
+        on the copied cases (useful for hiding sensitive data).
+
+            Parameters:
+                domain: the domain string
+                case_ids: case ids of the cases to copy
+                to_owner: the owner to copy the cases to
+                censor_data: the attributes/properties to be censored, specified as a dict.
+                    The keys corresponding to the property/attribute and the value specifies
+                    the type of censored data, i.e. number or date
+                count_only: specifies whether to return only the number of copied cases.
+
+            Returns:
+                The copied cases. If `count_only` is True only the count will be returned.
+        """
+        if not to_owner:
+            raise Exception('Must copy cases to valid new owner')
+
+        original_cases = CommCareCase.objects.get_cases(case_ids=case_ids, domain=domain)
+        if not any(original_cases):
```

Comment:
```suggestion
        if not original_cases:
```
... `any()` will check the truthiness of the first case, and I don't think you care about that.

Reply:
@kaapstorm What do you mean it "will check the truthiness of the first case"?

As far as I know it checks the truthiness of any item in the iterable, which would in this case either all be truthy or the list would be empty.

(I'm happy to change it, but I can't say I know why)

Reply:
Maybe an example:
```
>>> bool([])
False
>>> bool([0])
True
```
As you know, a list is truthy if it has items in it, regardless of the truthiness of the items.
```
>>> any([0])
False
>>> any([0, 1])
True
>>> all([0, 1])
False
>>> all([1])
True
```
`any()` and `all()` evaluate the truthiness of the items inside the list.

A case, as an object or a non-empty dictionary, will always be truthy, so `any()` will return `True` as soon as it encounters the first case. But we don't really need to check whether the case is truthy---we know it is---we just need to check that the list is not empty.
```
>>> case = object()
>>> any([case])
True
>>> bool([case])
True
```